### ISSUE TITLE
Handler system

### ISSUE THREAD

**Issue by 2Cubed** [CactusDev/CactusBot#44]
A class for all handlers to inherit from. Uses `on_` methods for event handling.

**Comment by Innectic** [comment]
This was finished on the `feature/handlers` branch.

**Comment by 2Cubed** [comment]
[Currently empty](https://github.com/CactusDev/CactusBot/blob/143370cb2cc0a833ea89b99a98d42e501b7ab02d/cactusbot/handler.py#L25). Need to add content, which will be possible once #43 is complete.

**Comment by Innectic** [comment]
@2Cubed I thought you said nothing needed to be added as of now?

**Comment by 2Cubed** [comment]
@Innectic Correct, but we shouldn't close the issue until we've actually added the things.

Something we should add, though, is a built-in logger.
```python
import logging

class Handler(object):
    """Abstract handler."""

    def __init__(self):
        self.logger = logging.getLogger(__name__)

class PotatoHandler(object):
    def on_potato(self, potato):
        self.logger.debug("We got a potato! (%s)", potato)
```

**Comment by Innectic** [comment]
I'll make that a thing real fast.

**Comment by Innectic** [comment]
@2Cubed Done in commit: [32d878aed050bdf502b9980f380370d1e620bbd6](https://github.com/CactusDev/CactusBot/commit/32d878aed050bdf502b9980f380370d1e620bbd6)

**Comment by 2Cubed** [comment]
@Innectic Whoa, nice! Only complaint is this:
```python
    def on_message(self, packet):
        """Handle message events."""
        self.logger.debug(packet)
        self.logger.info(
            ''.join(chunk["text"] for chunk in packet if chunk["type"] == chunk["text"])
        )
```

IMHO, we should keep the `LoggingHandler` (#48) and move those logging statements there.

**Comment by Innectic** [comment]
@2Cubed I don't see any use to keeping the logging handler if `Handler` contains a logger. Maybe you can prove me wrong?

**Comment by 2Cubed** [comment]
What else might this need? [Currently, it's essentially an empty class](https://github.com/CactusDev/CactusBot/blob/a898e3aa812406fb3f751085b68e0ef5157124fd/cactusbot/handler.py#L49).

**Comment by Innectic** [comment]
I'm not sure if anything is needed in there, at the moment.

**Comment by 2Cubed** [comment]
@Innectic Yeah, me neither. I feel like there should be *something*, but... :thinking:

**Comment by Innectic** [comment]
@2Cubed I think we could close this. (at least for now)

**Comment by 2Cubed** [comment]
@Innectic Renamed to `Handler system`, because that's a much better description of what this issue is for.

**Comment by 2Cubed** [comment]
Completed on `feature/handlers` branch, merged in #81.
